### PR TITLE
[frontend] Entities migration to tsx to prepare filters format refacto

### PR DESCRIPTION
**Pull request by Archidoit**
Migration of entities from js component to tsx pure functions to prepare the filters format refacto that will be done in #2686.
The migrations of the entities will enable to use hooks for those entities, instead of having to copy and paste the same code for managing filters (handleAddFilter, etc).

The concerned entities are:
- Notes 
- Search 
- Tools 
- Relationships
- Channels 
- Malwares 
- Entities 
- Events 
- Individuals 
- Organizations 
- Systems 
- Artifacts 
- AttackPatterns 
- CoursesOfAction 
- Campaigns 
- IntrusionSets
- ThreatActorsGroups

+ bug resolved: filter 'revoked' not working for Malwares and for CoursesOfAction

### CODE REVIEW

File: opencti-platform/opencti-front/src/private/components/analyses/notes/NoteLine.tsx [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+    height: 20,
+    float: 'left',
+    width: 120,
+  },
+}));
+
+interface NoteLineComponentProps {
+  node: NoteLine_node$data;
+  dataColumns: DataColumns;
+  onLabelClick: (
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/analyses/notes/NoteLine.tsx]: You can use the HandleAddFilter type

File: opencti-platform/opencti-front/src/private/components/arsenal/Malwares.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+          'x_opencti_workflow_id',
+          'labelledBy',
+          'markedBy',
+          'createdBy',
+          'source_reliability',
+          'confidence',
+          'created_start_date',
+          'created_end_date',
+          'malware_types',
+          'revoked',
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/arsenal/Malwares.tsx:81]: Not relative to your work, but this filter is not working (not present in enum back-end)

File: opencti-platform/opencti-front/src/private/components/arsenal/channels/ChannelLine.tsx [hunk 3]
```diff
@@ -33,26 +36,70 @@ const useStyles = makeStyles((theme) => ({
...
+    backgroundColor: theme.palette.grey?.[700],
   },
 }));
 
-const ChannelLineComponent = (props) => {
-  const { node, dataColumns, onLabelClick } = props;
+interface ChannelLineProps {
+  node: ChannelLine_node$key;
+  dataColumns: DataColumns;
+  onLabelClick: (
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/arsenal/channels/ChannelLine.tsx]: You can use this type HandleAddFilter

File: opencti-platform/opencti-front/src/private/components/observations/Artifacts.tsx [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+                    paginationOptions={paginationOptions}
+                    dataColumns={dataColumns}
+                    onLabelClick={helpers.handleAddFilter}
+                    selectedElements={selectedElements}
+                    deSelectedElements={deSelectedElements}
+                    onToggleEntity={onToggleEntity}
+                    selectAll={selectAll}
+                    setNumberOfElements={helpers.handleSetNumberOfElements}
+                  />
+            <ToolBar
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/observations/Artifacts.tsx]: The indentation is particular here

File: opencti-platform/opencti-front/src/private/components/observations/Artifacts.tsx [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+          </>
+    );
+  };
+
+  return (
+      <ExportContextProvider>
+        <div>
+          {renderLines()}
+          <Security needs={[KNOWLEDGE_KNUPDATE]}>
+            <ArtifactCreation
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/observations/Artifacts.tsx:189]: I can't create Artifact  (updater bug)

File: opencti-platform/opencti-front/src/private/components/techniques/CoursesOfAction.jsx [hunk 6]
```diff
@@ -1,197 +0,0 @@
...
-        availableFilterKeys={[
-          'x_opencti_workflow_id',
-          'labelledBy',
-          'markedBy',
-          'createdBy',
-          'source_reliability',
-          'creator',
-          'created_start_date',
-          'created_end_date',
-          'revoked',
```
Comment (RomuDeuxfois) [inline opencti-platform/opencti-front/src/private/components/techniques/CoursesOfAction.jsx:150]: This filter is not working

**Comment by RomuDeuxfois** [comment]
Tested 🆗 
A huge thanks for this work !!!

**Comment by Archidoit** [comment]
Thanks for the review @RomuDeuxfois 🌞  All the comments have been fixed.

**Comment by SamuelHassine** [comment]
Huge huge thanks Cathia for the work :)

**Review by RomuDeuxfois — APPROVED** [review]

**Pull request merged**